version: "3.7"

services:

  database:
    image: docker-registry.secure.amsterdam.nl/datapunt/bereikbaarheid-backend:${VERSION:-latest}
    ports:
      - "5409:5432"
    environment:
      POSTGRES_DB: bereikbaarheid
      POSTGRES_USER: bereikbaarheid
      POSTGRES_PASSWORD: insecure
    build:
      dockerfile: Dockerfile_pgrouting
    networks:
      - backend

  app: &app
    build:
      context: .
      target: app
    image: docker-registry.secure.amsterdam.nl/datapunt/bereikbaarheid-backend:${VERSION:-latest}
    volumes:
      - ./src:/app/src
      - ./tests:/app/tests
      - ./deploy:/deploy
    depends_on:
      - database

  dev:
    <<: *app
    build:
      target: dev
    image: docker-registry.secure.amsterdam.nl/datapunt/bereikbaarheid-backend-dev:${VERSION:-latest}
    environment:
      - DEBUG=true
      - SECRET_KEY=123456
      - DJANGO_SETTINGS_MODULE=main.settings
      - DATABASE_NAME=bereikbaarheid
      - DATABASE_USER=bereikbaarheid
      - DATABASE_PASSWORD=insecure
    command:
      python manage.py runserver 0.0.0.0:8000
    networks:
      - backend

  test:
    <<: *app
    build:
      target: tests
    image: docker-registry.secure.amsterdam.nl/datapunt/bereikbaarheid-backend-test:${VERSION:-latest}
    environment:
      - DEBUG=true
      - SECRET_KEY=123456
      - DJANGO_SETTINGS_MODULE=main.settings
      - DATABASE_NAME=bereikbaarheid
      - DATABASE_USER=bereikbaarheid
      - DATABASE_PASSWORD=insecure
    command:
      pytest
    networks:
      - backend

networks:
  backend:
    name: amsterdam-bereikbaarheid
